src/Data/Transport.elm

```diff
@@ -197,9 +211,19 @@ getTransportBetween scope impacts cA cB distances =
 
                         Nothing ->
                             -- reverse query source dict
-                            getTransportBetween scope impacts cB cA distances
+                            if stopRecursion then
+                                erroneous impacts
+
+                            else
+                                getTransportBetween True scope impacts cB cA distances
+                )
+            |> Maybe.withDefault
+                (if stopRecursion then
+                    erroneous impacts
+
+                 else
+                    getTransportBetween True scope impacts cB cA distances
                 )
```

Comment:
We should really rather have the two required checks performed at once and exposing the result in a `Result`:

```elm
getTransportBetween :
    Impacts
    -> Country.Code
    -> Country.Code
    -> Distances
    -> Result String Transport
getTransportBetween impacts cA cB distances =
    case
        ( distances |> Dict.get cA |> Maybe.andThen (Dict.get cB)
        , distances |> Dict.get cB |> Maybe.andThen (Dict.get cA)
        )
    of
        ( Just transport, _ ) ->
            Ok { transport | impacts = impacts }

        ( _, Just transport ) ->
            Ok { transport | impacts = impacts }

        ( Nothing, Nothing ) ->
            Err <|
                "No distance found between "
                    ++ Country.codeToString cA
                    ++ " et "
                    ++ Country.codeToString cB
```

If we really don't want to spend the required time to leverage a `Result`, the same versy simple approach can be used with a fallback:

```elm
getTransportBetween :
    Impacts
    -> Country.Code
    -> Country.Code
    -> Distances
    -> Transport
getTransportBetween impacts cA cB distances =
    case
        ( distances |> Dict.get cA |> Maybe.andThen (Dict.get cB)
        , distances |> Dict.get cB |> Maybe.andThen (Dict.get cA)
        )
    of
        ( Just transport, _ ) ->
            { transport | impacts = impacts }

        ( _, Just transport ) ->
            { transport | impacts = impacts }

        ( Nothing, Nothing ) ->
            default impacts
```